src/services/item/plugins/recycled/repository.ts

```diff
@@ -1,17 +1,41 @@
-import { In } from 'typeorm';
+import { EntityManager, In } from 'typeorm';
+import { QueryDeepPartialEntity } from 'typeorm/query-builder/QueryPartialEntity.js';
 
 import { PermissionLevel } from '@graasp/sdk';
 
-import { AppDataSource } from '../../../../plugins/datasource';
+import { MutableRepository } from '../../../../repositories/MutableRepository';
+import { DEFAULT_PRIMARY_KEY } from '../../../../repositories/const';
 import { Member } from '../../../member/entities/member';
 import { Item } from '../../entities/Item';
 import { RecycledItemData } from './RecycledItemData';
-import { CannotRestoreNonDeletedItem } from './errors';
 
-export const RecycledItemDataRepository = AppDataSource.getRepository(RecycledItemData).extend({
-  async getOwnRecycledItemDatas(member: Member): Promise<RecycledItemData[]> {
+type CreateRecycledItemDataBody = { itemPath: string; creatorId: string };
+
+export class RecycledItemDataRepository extends MutableRepository<RecycledItemData, never> {
+  constructor(manager?: EntityManager) {
+    super(DEFAULT_PRIMARY_KEY, RecycledItemData, manager);
+  }
+
+  // warning: this call insert in the table
+  // but does not soft delete the item
+  // should we move to core item?
+  async addOne({ itemPath, creatorId }: CreateRecycledItemDataBody) {
+    return await super.insert({ item: { path: itemPath }, creator: { id: creatorId } });
+  }
+
+  // warning: this call insert in the table
+  // but does not soft delete the item
+  // should we move to core item?
+  async addMany(items: Item[], creator: Member) {
+    const recycled = items.map((item) => this.repository.create({ item, creator }));
+    this.repository.insert(recycled as QueryDeepPartialEntity<RecycledItemData>[]);
```

Comment:
Is the casting fixed by an update of the sdk ? I think I removed the `extra.app.setting: unknown` property last week ?